src/Faithlife.Tracing.Zipkin/ZipkinTracer.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Collections.Generic;
+using zipkin4net;
+using zipkin4net.Propagation;
+using zipkin4net.Tracers.Zipkin;
+using zipkin4net.Transport.Http;
+
+namespace Faithlife.Tracing.Zipkin
+{
+	public sealed class ZipkinTracer : ITracer
+	{
+		public static ITracer Create(Uri collectorUri) => collectorUri == null ? null : new ZipkinTracer(collectorUri);
+
+		private ZipkinTracer(Uri collectorUri)
+		{
+			IZipkinSender sender;
+			ISpanSerializer serializer;
+
+			if (collectorUri.Scheme == "http" || collectorUri.Scheme == "https")
+			{
+				sender = new HttpZipkinSender(collectorUri.AbsoluteUri, "application/json");
+				serializer = new JSONSpanSerializer();
+			}
+			else
+			{
+				throw new ArgumentException("Unrecognized Zipkin endpoint", nameof(collectorUri));
+			}
+
+			TraceManager.SamplingRate = 1.0f;
+			TraceManager.Trace128Bits = true;
+			var tracer = new zipkin4net.Tracers.Zipkin.ZipkinTracer(sender, serializer, new Statistics());
+			TraceManager.RegisterTracer(tracer);
+			TraceManager.Start(new ZipkinLog4netLogger());
+		}
+
+		public ITrace ExtractTrace(Func<string, string> extractKey)
+		{
+			var traceContext = s_extractor.Extract(extractKey);
```

Comment:
Without objection, will change this to attempt to extract both W3C and B3. (I anticipate that this would let us support being instrumented when being called from a Node.js API proxy that uses `zipkin-instrumentation-fetch`, e.g., see https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin/src/httpHeaders.js.)